### DISCUSSION TITLE
正文引用一篇多作者的英文文献时如何用“等”代替“et al.”?

### DISCUSSION THREAD

**Discussion by KKGeng** [whutug/whu-thesis#222] [Q&A]
您好，我在正文引用一篇多作者的英文文献时，编译后其余作者显示“et al.”，例如“Sun et al. (2015)”。

当修改gbt7714-2005-author-year.bst文件中FUNCTION {bbl.et.al}下的 { "et~al." }为{ "等" }以后，即
FUNCTION {bbl.et.al}
{ entry.lang lang.zh =
    { "等" }
    { entry.lang lang.ja =
        { "他" }
        { entry.lang lang.ru =
            { "идр" }
            { "等" }
            %%{ "et~al." }
          if$
        }
      if$
    }
  if$
}
虽然正文在引用时会变为“Sun 等 (2015)”，但是论文后的参考文献中的“et al.”也会对应变为“等”。

请问如何保证在正文引用文献时，始终用“等”，而在正文后的参考文献中，中文用“等”，英文用“et al.”

**Comment by sikouhjw** [comment]
bibtex 不好做，建议你用 biblatex-gb7714-2015 宏包的 `\setlocalbibstring` 功能。

**Reply by KKGeng** [reply]
感谢回复，我去试一下。

**Reply by KKGeng** [reply]
我发现一种解决我所提出问题的办法，只需要将gbt7714-2005-author-year.bst文件中第66行的
#0 'terms.in.macro :=
改为
#1 'terms.in.macro :=
便可以实现正文引用文献强制都用“等”，而后文中文文献用“等”，英文文献用“et al.”
